### ISSUE TITLE
Support for trio

### ISSUE THREAD

**Issue by kerrickstaley** [johnpaulett/python-hl7#33]
Hi! I'm starting work on a project which takes in HL7 data with `python-hl7` and writes that data to a database. To support concurrency, I was planning on using the `threading` stdlib module, but since `python-hl7` and `txHL7` don't provide out-of-the-box support for synchronous (blocking) networking (correct me if I'm wrong), I started looking at async options instead.

It seems like [`trio`](https://github.com/python-trio/trio) is gaining a lot of support as a simpler async implementation that avoids pitfalls in `asyncio` and leads to shorter, more readable code. The author of `trio` has a few articles ([1](https://stackoverflow.com/questions/49482969/what-is-the-core-difference-between-asyncio-and-trio) [2](https://vorpus.org/blog/some-thoughts-on-asynchronous-api-design-in-a-post-asyncawait-world/)) that explain how it makes asynchronous code easier to write.

[`trio-asyncio`](https://github.com/python-trio/trio-asyncio) is a compatibility layer that allows using `asyncio` libraries in `trio` applications, but it seems like it's not as easy as using a library that has native `trio` support. [`anyio`](https://anyio.readthedocs.io/en/latest/) is a library that lets other libraries natively support `trio`, `asyncio`, and `curio` (an older library similar to `trio`).

Given that you're actively working on an `asyncio` support for `python-hl7`, I was wondering what your thoughts are on `trio` support. I think there are a couple possible ways to proceed (any of these is OK, just wondering what you think):
- Support only `asyncio`; `trio` users can use `trio-asyncio`.
- Support `trio`, `asyncio`, and `curio` by using `anyio`.
- Support only `trio`. May actually save you time and make it easier to write the MLLP feature given `trio`'s emphasis on simplicity and usability, but would limit which projects can use this feature.

**Comment by johnpaulett** [comment]
To your first question about blocking & threading: correct, neither txHL7 or the new hl7.mllp are blocking. There is an example of using [deferredToThread in txHL7](https://txhl7.readthedocs.io/en/latest/custom-receiver.html#deferring-to-a-thread) that may be of interest to you. I expect that a similar behavior using ThreadPoolExecutor or ProcessPoolExecutor could be used in hl7.mllp [1](https://pymotw.com/3/asyncio/executors.html) [2](https://stackoverflow.com/questions/53158101/how-can-i-synchronize-asyncio-with-other-os-threads), and would welcome documentation examples of that.

To date, I have tried to keep python-hl7 dependency-free, which is part of the reason why txHL7 was a separate project and why we pursued the stdlib-based asyncio approach.

I'll admit I lack practical experience with trio, so the benefit vs asyncio is not immediately apparent (especially as asyncio has matured since the Python 3.4 release).

At this time, I'd be more open to:
1. A documentation example that shows usage of trio-asyncio to integrate hl7.mllp
2. PRs that refactor the hl7.mllp module to expose reusable code for alternative implementations (trio, txHL7, etc)
